File: Python/problem0747.py
class Solution:
    def dominantIndex(self, nums: List[int]) -> int:
        if not nums:
            return -1
        if len(nums) == 1:
            return 0
        d = {}
        for k in range(len(nums)):
            d[nums[k]] = k
        nums.sort(reverse=True)
        if nums[1] == 0:
            if nums[0] != 0:
                return d[nums[0]]
            else:
                return -1
        if nums[0] >= 2*nums[1]:
            return d[nums[0]]
        else:
            return -1

File: C++/problem0747.cpp
class Solution {
public:
    int dominantIndex(vector<int>& nums) {
        if (nums.size() == 1)
        {
            return 0;
        }
        
        int First = INT_MIN, Second = INT_MIN, first = -1/*, second = -1*/;
        for (vector<int>::iterator it = nums.begin(); it < nums.end(); ++it)
        {
            if (*it > First)
            {
                Second = First/*, second = first*/;
                First = *it, first = it - nums.begin();
            }
            else if (*it > Second)
            {
                Second = *it/*, second = it - nums.begin()*/;
            }
        }

        if (First >= Second * 2)
        {
            return first;
        }
        else
        {
            return -1;
        }
    }
};
